src/ARMJIT_Memory.cpp

```diff
@@ -628,8 +640,14 @@ void Init()
     FastMem7Start = mmap(NULL, AddrSpaceSize, PROT_NONE, MAP_ANON | MAP_PRIVATE, -1, 0);
 
     MemoryBase = (u8*)mmap(NULL, MemoryTotalSize, PROT_NONE, MAP_ANON | MAP_PRIVATE, -1, 0);
+    #ifdef __APPLE__
+        char* fastmem_pid_name = new char[snprintf(NULL, 0, "melondsfastmem%d", getpid())];
```

Comment:
apparently the value returned by snprintf doesn't take the null terminator into account, so it needs to be increased by one